src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,218 @@
+
+            protected static IInvocationOperation GetInvocation(IBinaryOperation violation)
+            {
+                var result = UseStringEqualsOverStringCompare.GetInvocationFromEqualityCheckWithLiteralZero(violation);
+
+                RoslynDebug.Assert(result is not null);
+
+                return result;
+            }
+
+            protected static SyntaxNode InvertIfNotEquals(SyntaxNode stringEqualsInvocationExpression, IBinaryOperation equalsOrNotEqualsOperation, SyntaxGenerator generator)
+            {
+                return equalsOrNotEqualsOperation.OperatorKind is BinaryOperatorKind.NotEquals ?
+                    generator.LogicalNotExpression(stringEqualsInvocationExpression) :
+                    stringEqualsInvocationExpression;
+            }
+        }
+
+        /// <summary>
+        /// Replaces <see cref="string.Compare(string, string)"/> violations.
+        /// </summary>
+        private sealed class StringStringCaseReplacer : OperationReplacer
+        {
+            public StringStringCaseReplacer(RequiredSymbols symbols)
+                : base(symbols)
+            { }
+
+            public override bool IsMatch(IBinaryOperation violation) => UseStringEqualsOverStringCompare.IsStringStringCase(violation, Symbols);
+
+            public override SyntaxNode CreateReplacementExpression(IBinaryOperation violation, SyntaxGenerator generator)
+            {
+                RoslynDebug.Assert(IsMatch(violation));
+
+                var compareInvocation = GetInvocation(violation);
+                var equalsInvocationSyntax = generator.InvocationExpression(
+                    CreateEqualsMemberAccess(generator),
+                    compareInvocation.Arguments.GetArgumentsInParameterOrder().Select(x => x.Value.Syntax));
+
+                return InvertIfNotEquals(equalsInvocationSyntax, violation, generator);
+            }
+        }
+
+        /// <summary>
+        /// Replaces <see cref="string.Compare(string, string, bool)"/> violations.
+        /// </summary>
+        private sealed class StringStringBoolReplacer : OperationReplacer
+        {
+            public StringStringBoolReplacer(RequiredSymbols symbols)
+                : base(symbols)
+            { }
+
+            public override bool IsMatch(IBinaryOperation violation) => UseStringEqualsOverStringCompare.IsStringStringBoolCase(violation, Symbols);
+
+            public override SyntaxNode CreateReplacementExpression(IBinaryOperation violation, SyntaxGenerator generator)
+            {
+                RoslynDebug.Assert(IsMatch(violation));
+
+                var compareInvocation = GetInvocation(violation);
+
+                //  We know that the 'ignoreCase' argument in 'string.Compare(string, string, bool)' is a boolean literal
+                //  because we've asserted that 'IsMatch' returns true.
+                var ignoreCaseLiteral = (ILiteralOperation)compareInvocation.Arguments.GetArgumentForParameterAtIndex(2).Value;
```

Comment:
Is this also true for cases where a boolean variable is used? I haven't checked `UseStringEqualsOverStringCompare.IsStringStringBoolCase` to know if it handles that as well

```c# 
bool CompareNames(Person p, bool ignoreCase) => string.Compare(Name, p.Name, ignoreCase)
```

Reply:
The analyzer only flags cases where the boolean argument is a literal.